src/snowflake/cli/api/cli_global_context.py

```diff
@@ -399,19 +274,44 @@ def _should_force_mute_intermediate_output(self) -> bool:
         return self._manager.output_format == OutputFormat.JSON
 
 
-_CLI_CONTEXT_MANAGER: _CliGlobalContextManager | None = None
-_CLI_CONTEXT: _CliGlobalContextAccess | None = None
+_CLI_CONTEXT_MANAGER: ContextVar[_CliGlobalContextManager] = ContextVar(
+    "cli_context", default=_CliGlobalContextManager()
+)
 
 
 def get_cli_context_manager() -> _CliGlobalContextManager:
-    global _CLI_CONTEXT_MANAGER
-    if _CLI_CONTEXT_MANAGER is None:
-        _CLI_CONTEXT_MANAGER = _CliGlobalContextManager()
-    return _CLI_CONTEXT_MANAGER
+    return _CLI_CONTEXT_MANAGER.get()
 
 
 def get_cli_context() -> _CliGlobalContextAccess:
-    global _CLI_CONTEXT
-    if _CLI_CONTEXT is None:
-        _CLI_CONTEXT = _CliGlobalContextAccess(get_cli_context_manager())
-    return _CLI_CONTEXT
+    return _CliGlobalContextAccess(get_cli_context_manager())
+
+
+@contextmanager
+def fork_cli_context(
+    connection_overrides: dict | None = None,
+    env: dict[str, str] | None = None,
+    project_is_optional: bool | None = None,
+    project_path: str | None = None,
+) -> Iterator[_CliGlobalContextAccess]:
+    """
+    Forks the global CLI context, making changes that are only visible
+    (e.g. via get_cli_context()) while inside this context manager.
+    """
+    new_manager = _CLI_CONTEXT_MANAGER.get().clone()
+    token = _CLI_CONTEXT_MANAGER.set(new_manager)
```

Comment:
Why it's called token?

Reply:
https://docs.python.org/3/library/contextvars.html#contextvars.Token